var/spack/repos/builtin/packages/gunrock/package.py

```diff
@@ -23,9 +26,100 @@ class Gunrock(CMakePackage, CudaPackage):
+    variant('mgpu_tests',           default=False, description='Builds Gunrock applications and enables the ctest framework for single GPU implementations')
+    variant('cuda_verbose_ptxas',   default=False, description='Enable verbose output from the PTXAS assembler')
+    variant('google_tests',         default=False, description='Build unit tests using googletest')
+    variant('code_coverage',        default=False, description="run code coverage on Gunrock's source code")
+    # apps
+    msg = 'select either all or individual applications'
+    variant(
+        "applications",
+        values=disjoint_sets(
+            ('all',), ('bc', 'bfs', 'cc', 'pr', 'sssp', 'dobfs', 'hits',
+                       'salsa', 'mst', 'wtf', 'topk')
+        ).allow_empty_set().with_default('all').with_error(msg),
+        description="Application to be built"
+    )
+
+    variant('boost', default=True, description='Build with Boost')
+    variant('metis', default=False, description='Build with Metis support')
+
+    depends_on('googletest', when='+google_tests')
+    depends_on('lcov', when='+code_coverage')
+    depends_on('boost@1.58.0:', when='+boost')
+    depends_on('metis', when='+metis')
+
+    conflicts('cuda_arch=none', when='+cuda',
+              msg='Must specify CUDA compute capabilities of your GPU. \
+See "spack info gunrock"')
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+        args.extend([
+                    '-DGUNROCK_BUILD_LIB={0}'.format(
+                        'ON' if '+lib' in spec else 'OFF'),
+                    '-DGUNROCK_BUILD_SHARED_LIBS={0}'.format(
+                        'ON' if '+shared_libs' in spec else 'OFF'),
+                    '-DGUNROCK_BUILD_TESTS={0}'.format(
+                        'ON' if '+tests' in spec else 'OFF'),
+                    '-DGUNROCK_MGPU_TESTS={0}'.format(
+                        'ON' if '+mgpu_tests' in spec else 'OFF'),
+                    '-DCUDA_VERBOSE_PTXAS={0}'.format(
+                        'ON' if '+cuda_verbose_ptxas' in spec else 'OFF'),
+                    '-DGUNROCK_GOOGLE_TESTS={0}'.format(
+                        'ON' if '+google_tests' in spec else 'OFF'),
+                    '-DGUNROCK_CODE_COVERAGE={0}'.format(
+                        'ON' if '+code_coverage' in spec else 'OFF'),
+                    ])
+
+        # turn off auto detect, which undoes custom cuda arch options
+        args.append('-DCUDA_AUTODETECT_GENCODE=OFF')
+
+        cuda_arch_list = self.spec.variants['cuda_arch'].value
+        if cuda_arch_list[0] != 'none':
+            for carch in cuda_arch_list:
+                args.append('-DGUNROCK_GENCODE_SM' + carch + '=ON')
+
+        app_list = self.spec.variants['applications'].value
+        if app_list[0] != 'none':
+            args.extend([
+                '-DGUNROCK_BUILD_APPLICATIONS={0}'.format(
+                    'ON' if 'all'           in app_list else 'OFF'),
```

Comment:
The more idiomatic way to do the same would be:
```suggestion
                    'ON' if spec.satisfies('applications=all') else 'OFF'),
```
Can you check if this works? In this way the package would rely only on the spec syntax and not on implementation details that might change in the future.

Reply:
I think this only works for `all` value but not others since user might specify `applications=bc,bfs` then `'ON' if spec.satisfies('applications=bc') else 'OFF'), ` won't work I think.

Reply:
Actually ... it does work. so, will implement it for all apps options.

Reply:
> Actually ... it does work. so, will implement it for all apps options.

Yes, `applications=bc,bfs` satisfies both `applications=bc` and `application=bfs`